Add Writing guidelines

gchaps opened the pull request:
Provides descriptions and examples of how to write UI text

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        <EuiPanel paddingSize="l">
+          <EuiText>
+            <h3>Clear and concise</h3>
+            <p>Get straight to the point&mdash;in a that way your users understand.  Make every word contribute to meaning.</p>
```
"in a that way" -- typo?

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+
+    <EuiSpacer />
+    <EuiSpacer />
+    <EuiSpacer />
```
Instead of using three spacers, you can use a single extra-large one:

```jsx
<EuiSpacer size="xxl" />
```

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+
+      <EuiFlexItem>
+        <EuiToast color="success"  >
+          <p>Didn&apos;t find what you were looking for?</p>
```
What do you think of adding a "don't" to contrast this with? I think it will make it easier for people to see how the lack of a contraction sounds awkward: "Did not find what you were looking for?"

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+
+      <EuiFlexItem>
+        <EuiToast color="success"  >
+          <p>You don&apos;t have any dashboards. Let&apos;s create some!</p>
```
I think a contrasting "don't" example here could also make it clear how too many exclamation points sounds weird. What do you think?

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        <EuiText>
+          <p>
+            <strong>Summarize the message in the title.</strong>&nbsp;
+            Don&rsquo;t use the words error, warning, confirm, or jaron such as oops and uh-oh.
```
Typo: "jaron" ->  "jargon"

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+          color="success"
+        >
+        For maximum compatibility, share the short URL of the snapshot.
+        Not all wiki and markup parsers can the handle full-length version.
```
"parsers can the handle" -> "parsers can handle the"

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        <EuiToast
+          color="success"
+        >
+      checkbox with label Show remaining slices
```
We can make this a checkbox like this:

```jsx
<EuiCheckbox
  onChange={() => {}}
  id="checkbox1"
  label="Show remaining slices"
/>
```

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        <EuiToast
+          color="danger"
+        >
+        checkbox with label Show other
```
And here:

```jsx
<EuiCheckbox
  onChange={() => {}}
  id="checkbox2"
  label="Show other"
/>
```

cchaos commented:
Can we put the "Do"s and "Don't"s inside the toasts instead of below them?

<img width="656" alt="region capture 4" src="https://user-images.githubusercontent.com/549577/34055922-3b755c6a-e19f-11e7-9c3b-a34545437174.png">

snide commented:
@cchaos I'm planning on doing a rework of the design of this page. Told @gchaps to just focus on the content. You and I can clean help clean it up after its merged. For example, we shouldn't be using toasts for this stuff, since that's for higher z-index, ephemeral content.

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+      <EuiFlexItem>
+        <EuiToast color="success"  >
+                  Number
+          <EuiFieldText />
```
We should use the correct implementation of form elements and their labels and help text like:
```ts
<EuiFormRow
  id={makeId()}
  label="Number"
  helpText="Number must be between 1 and 5"
>
  <EuiFieldText name="num" />
</EuiFormRow>
```

@gchaps Let me know if you want help doing that.

snide commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+    <EuiSpacer />
+
+    <EuiText>
+      <h2>Capitalization</h2>
```
I'd add an example in here that says when capitalization is OK (like proper nouns and product names). Usually people forget something like "Machine Learning" is a product of and a proper noun.

Also, I could be missing our usage here completely! Which is why it might be good to clarify it in more detail.

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+      <EuiFlexItem>
+        <EuiToast
+          title="Uh-oh!"
+          color="success"
```
This toast color should be danger.

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,810 @@ import {
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiText>
+          <p><strong>Avoid using &quot;please.&quot;</strong>&nbsp;
```
**Yes!** 🥇

snide commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        <EuiToast color="success"  >
+                  Number
+          <EuiFieldText />
+                  Number must be between 1 and 5
```
Note to myself for later. This should use the `<FieldRow>` component. 

Question for @gchaps though. Specifically for help text, what if you need more than two sentences? Should we say multi-sentence help text isn't allowed?

gchaps replied:
@snide The UI should layer information.  Do we have a component like signpost help?  I can also add something about linking to documentation

snide commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiText>
+          <p><strong>Use contractions</strong> if they make your text easier to comprehend.</p>
```
Again, I'd link to some primer on what contractions are. People might not know.

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,810 @@ import {
+      <EuiFlexItem>
+        <EuiText>
+          <p>
+            <strong>Avoid the urge to explain
```
For the example of this one, can we show some self-explanatory fields that don't require help text? Then add another row using the example from this current one that talks about being too verbose in help text?

cchaos replied:
@gchaps ^^ Mainly just this comment needs your attention

gchaps replied:
@cchaos Sure I can add this information .  Nothing bugs me more than seeing a form with fields for Name and Location and then showing placeholder text that says the same thing.

gchaps commented:
I fixed the typos and added the code for the checkbox. Thanks, @cjcenizal 

The reason the Dos and Don'ts are at the bottom is so that the reader focuses on the text first, instead of the Dos and Don'ts.  

@cchaos and @snide looking forward to how you improve the design.  Let me know if you have any comments about the content.

snide commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+        >
+        Password
+          <EuiFieldPassword defaultValue="password" />
+        Must be least 8 characters and include upper and lower case letters, numbers, and symbols such as !@#$%&.
```
Again, remind for myself to convert all this stuff to `FormRow`.

I notice these help-text use punctuation. Intentional? Seems to contradict your earlier rule.

FWIW I think no punctuation is good for titles and statements, but I like them for form-label help text. Specifically because they often end up multi-sentence.

gchaps commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,807 @@ import {
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiText>
+          <p><strong>Use active rather than passive voice.</strong> It&apos;s more clear and less wordy.</p>
```
@snide I'll give some quick primers on contractions, sentence case, and active vs. passive voice--or link to them.  Also thanks for cleaning up the components--I know that the buttons in the mock dialog boxes are on the wrong side.

gchaps commented:
Ready for another review.

cjcenizal commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,992 @@ import {
+          </p>
+        </EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem>
```
Can we find a different example for this point? Labels are really important usability and accessibility aids for forms. They communicate the role of an input even after the placeholder text has gone (e.g. when the user has started typing), and screen readers use them to tell users what the input is for. I advocate for preserving labels in general. Maybe we can use an example which has unnecessary help text below the input instead?

cchaos commented:
One quick overall question, are we ok with having Do examples without a corresponding Don't and vice versa?

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,992 @@ import {
+            size="s"
+            onClick={() => window.alert('Button clicked')}
+          >
+          Remove index
```
I would think this should just be "Remove" not "Remove index"?

cchaos commented on src-docs/src/views/guidelines/writing.js:
```diff
@@ -4,20 +4,992 @@ import {
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiText>
+          <h3>Don&apos;t be clever wtih a serious message</h3>
```
Spelling - wtih -> with

cjcenizal commented:
> are we ok with having Do examples without a corresponding Don't and vice versa?
Personally I really appreciate the contrast of a Don't with every Do. It helps me zero in on the point more quickly.

gchaps commented:
Got it.  I've added a few more don'ts
